plugin/pkg/admission/runtimeclass/admission.go

```diff
@@ -171,13 +183,39 @@ func setOverhead(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta1.Run
 	return nil
 }
 
-func validateOverhead(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta1.RuntimeClass) (err error) {
+func setScheduling(a admission.Attributes, pod *api.Pod, runtimeClass *v1beta1.RuntimeClass) (err error) {
+	if runtimeClass == nil || runtimeClass.Scheduling == nil {
+		return nil
+	}
 
+	// convert to internal type and assign to pod's Scheduling
+	nodeScheduling := &node.Scheduling{}
+	if err = nodev1beta1.Convert_v1beta1_Scheduling_To_node_Scheduling(runtimeClass.Scheduling, nodeScheduling, nil); err != nil {
+		return err
+	}
+
+	runtimeNodeSelector := nodeScheduling.NodeSelector
+	newNodeSelector := pod.Spec.NodeSelector
+	for key, value := range runtimeNodeSelector {
+		if podValue, ok := newNodeSelector[key]; ok && podValue != value {
+			return admission.NewForbidden(a, fmt.Errorf("pod rejected: RuntimeClass Scheduling's NodeSelector %q is conflict with Pod's NodeSelector", key))
+		}
+		newNodeSelector[key] = value
+	}
+
+	newTolerations := tolerations.MergeTolerations(pod.Spec.Tolerations, nodeScheduling.Tolerations)
```

Comment:
Merge tolerations is incomplete, since it only keys off of key & effect, but it's valid to have different values and/or operators. I suggest using a custom method here, and then following up with a separate PR to fix MergeTolerations (if you're up for it).

On a related note, we need to decide what to do if there is a conflict (same {key, op, val, effect} and different `TolerationSeconds`). I think options are: use the pod value, return an error (reject the pod), or update the API validation to forbid setting TolerationSeconds on RuntimeClass (punt the problem)

Reply:
Could you explain a little bit on how to merge the tolerations? I follow the KEP to use this utility function to merge them.

> This means that RuntimeClass tolerations can simply be appended to the existing tolerations, but an [existing utilty](https://github.com/kubernetes/kubernetes/blob/58021216b16ae6d5f24fb1c32ab541b2e79a365e/pkg/util/tolerations/tolerations.go#L62) can reduce duplicates by merging equivalent tolerations.

Reply:
The PodRestrictionTolerations use the same utility function, is it on purpose to override the toleration in these situations?

Reply:
Fixed in https://github.com/kubernetes/kubernetes/pull/81732